keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/BakeryLifecycleMonitor.kt

```diff
@@ -57,25 +60,15 @@ class BakeryLifecycleMonitor(
           else -> publishUnknownEvent(task, execution)
         }
 
-        if (task.numFailures > 0) {
-          // we only care about consecutive failures, and we just had a success
-          monitorRepository.clearFailuresGettingStatus(task)
-        }
-
         if (execution.status.isComplete()) {
           endMonitoringOf(task)
+        } else {
+          markSuccessFetching(task)
         }
       }
       .onFailure { exception ->
         log.error("Error fetching status for $task: ", exception)
-        if (task.numFailures >= lifecycleConfig.numFailuresAllowed - 1) {
-          log.warn("Too many consecutive errors (${lifecycleConfig.numFailuresAllowed}) " +
-            "fetching the task status for $task. Giving up.")
-          publishExceptionEvent(task)
-          monitorRepository.delete(task)
-        } else {
-          monitorRepository.markFailureGettingStatus(task)
-        }
+        handleFailureFetching(task)
```

Comment:
Same comment on the method name here.

Reply:
To me, handle implies it's doing more, which that method is doing.